src/rdb_protocol/order_util.cc

```diff
@@ -33,7 +33,7 @@ build_comparison(const term_t* target,
 
     counted_t<const func_t> comparison_function = arg->as_func(GET_FIELD_SHORTCUT);
     rcheck_target(target,
-                  comparison_function->is_deterministic() != deterministic_t::no,
+                  (comparison_function->is_deterministic() &~ DET_SINGLE_SERVER) == DET_DETERMINISTIC,
```

Comment:
So this was saying the value is either single-server or deterministic.

But now in a convoluted way, it's saying that the value is either DET_SINGLE_SERVER or DET_DETERMINISTIC, but not DET_CONSTANT_NOW.